submodules/telemetry/__init__.py

```diff
@@ -1,133 +1,162 @@
+        """
+        if not ((type(message) is str and message[0] == ';') or type(message) is error.Error or type(
+                message) is log.Log):   # check for valid types, error if invalid
+            self.logger.error("Attempted to enqueue invalid message")
+            return False
+        with self.packet_lock:
+            self.general_queue.append(message)  # append to general queue
+            return True
+
+    def dump(self, radio='aprs') -> bool:
+        """
+        Concatenates packets to fit in max_packet_size (defined in config) and send through the radio, removing the
+        packets from the error and log stacks in the process
+        :param radio: Radio to send telemetry through, either "aprs" or "iridium"
+        :return True if anything was sent, false otherwise
+        """
+        squishedpackets = ""
+        retVal = False
+
+        if not self.has_modules:
+            raise RuntimeError("self.modules empty and not initialized")
+
+        with self.packet_lock:
+            while len(self.log_stack) + len(self.err_stack) > 0:    # while there's stuff to pop off
+                next_packet = (str(self.err_stack[-1]) if len(self.err_stack) > 0 else str(self.log_stack[-1]))   # for the purposes of determining packet length
+                while len(base64.b64encode((squishedpackets + next_packet).encode('ascii'))) < self.config["telemetry"]["max_packet_size"] and len(self.log_stack) + len(self.err_stack) > 0:
+                    if len(self.err_stack) > 0: # prefer error messages over log messages
+                        squishedpackets += str(self.err_stack.pop())
+                    else:
+                        squishedpackets += str(self.log_stack.pop())
+                squishedpackets = base64.b64encode(squishedpackets.encode('ascii'))
+                # print(squishedpackets)
+                self.modules[radio].send(str(squishedpackets))
+                retVal = True
+                squishedpackets = ""
+
+        return retVal
+
+    def clear_buffers(self) -> None:
+        """
+        Clear the telemetry buffers - clearing general_queue, the log, and error stacks.
+        :return: None
+        """
+        with self.packet_lock:
+            self.general_queue.clear()
+            self.log_stack.clear()
+            self.err_stack.clear()
+
+    def decide(self) -> None:
+        """
+        A thread method to constantly check general_queue for messages and process them if there are any.
+        :return: None
+        """
+        while True:
+            if len(self.general_queue) != 0:
+                if not self.has_modules:
+                    raise RuntimeError("self.modules empty and not initialized")
+                with self.packet_lock:
+                    message = self.general_queue.popleft()
+                    if type(message) is str and message[0] == ';':
+                        self.modules["command_ingest"].enqueue(message)
```

Comment:
call `has_modules` here to ensure "command_ingest" is found
if there are no modules `raise RuntimeError`